src/impl_owned_array.rs

```diff
@@ -59,3 +64,143 @@ where
         self.data.into_vec()
     }
 }
+
+/// Methods specific to `Array2`.
+///
+/// ***See also all methods for [`ArrayBase`]***
+///
+/// [`ArrayBase`]: struct.ArrayBase.html
+impl<A> Array<A, Ix2> {
+    /// Append a row to an array with row major memory layout.
+    ///
+    /// ***Errors*** with a layout error if the array is not in standard order or
+    /// if it has holes, even exterior holes (from slicing). <br>
+    /// ***Errors*** with shape error if the length of the input row does not match
+    /// the length of the rows in the array. <br>
+    ///
+    /// The memory layout matters, since it determines in which direction the array can easily
+    /// grow. Notice that an empty array is compatible both ways. The amortized average
+    /// complexity of the append is O(m) where *m* is the length of the row.
+    ///
+    /// ```rust
+    /// use ndarray::{Array, ArrayView, array};
+    ///
+    /// // create an empty array and append
+    /// let mut a = Array::zeros((0, 4));
+    /// a.try_append_row(ArrayView::from(&[1., 2., 3., 4.])).unwrap();
+    /// a.try_append_row(ArrayView::from(&[0., -2., -3., -4.])).unwrap();
+    ///
+    /// assert_eq!(
+    ///     a,
+    ///     array![[1., 2., 3., 4.],
+    ///            [0., -2., -3., -4.]]);
+    /// ```
+    pub fn try_append_row(&mut self, row: ArrayView<A, Ix1>) -> Result<(), ShapeError>
```

Comment:
Consider using the name `append_row`?

Reply:
The input could be `&[A]` or ArrayView, but it could also be a `Vec<A>` or `Array1<A>` for ownership semantics. Doesn't look possible to cover both in the same method in a simple way.

Reply:
In principle, it would be possible to cover both the owned and non-owned cases by accepting `row: ArrayBase<S2, Ix1> where S2: Data<Elem = A>` and adding the necessary functionality to `Data` to either clone or move elements based on the ownership of the data. I haven't thought through the details of how difficult this would be to implement, though.

Reply:
No action on this IMO